### PR TITLE
feat: add new solution for question 1.2

### PR DESCRIPTION
**Pull request by damla**
## Solution 1

- One line solution added as a more competitive choice.

## Solution 2

- Done with hashMap as a more optimized solution.

### CODE REVIEW

File: chapter01/1.2 - Check Perm/checkPermute.js [hunk 1]
```diff
@@ -16,4 +16,12 @@ console.log(checkPermute('aba', 'aab'), true);
...
\ No newline at end of file
+console.log(checkPermute('aba', 'aa'), false);
+
+// Same function as a one line solution
+
+checkPermuteOneLine = (a, b) => {
+    return (a.length === b.length && (a.split("").sort().join("") === b.split("").sort().join("")))
+}
+
+console.log(checkPermuteOneLine('aba', 'baa'));
```
Comment (damla) [inline chapter01/1.2 - Check Perm/checkPermute.js:27]:
```suggestion
// Tests
console.log(checkPermuteOneLine('aba', 'baa'), true);
console.log(checkPermuteOneLine('aba', 'aa'), false);

// HashMap Solution

isPermutation = (a, b) => {

    if (a.length !== b.length) return false;

    let map = new Map();

    for (let i = 0; i < a.length; i++) {
        if (!map.get(a[i])) map.set(a[i], 0);
        map.set(a[i], map.get(a[i]) + 1);
    }
    for (let i = 0; i < b.length; i++) {
        map.set(b[i], map.get(b[i]) - 1);
        if (map.get(b[i]) < 0) {
            return false;
        }
    }
    return true;
}

console.log(isPermutation("damla", "almad"), true);
console.log(isPermutation("damla", "dmla"), false);
console.log(isPermutation("damla", "daala"), false);
```

File: chapter01/1.2 - Check Perm/checkPermute.js [hunk 2]
```diff
@@ -16,4 +16,39 @@ console.log(checkPermute('aba', 'aab'), true);
...
+checkPermuteOneLine = (a, b) => {
+    return (a.length === b.length && (a.split("").sort().join("") === b.split("").sort().join("")))
+}
+
+// Tests
+console.log(checkPermuteOneLine('aba', 'baa'), true);
+console.log(checkPermuteOneLine('aba', 'aa'), false);
+
+// HashMap Solution
+
```
Comment (damla) [inline chapter01/1.2 - Check Perm/checkPermute.js:32]:
```suggestion
```

File: chapter01/1.2 - Check Perm/checkPermute.js [hunk 3]
```diff
@@ -16,4 +16,39 @@ console.log(checkPermute('aba', 'aab'), true);
...
+    }
+    for (let i = 0; i < b.length; i++) {
+        map.set(b[i], map.get(b[i]) - 1);
+        if (map.get(b[i]) < 0) {
+            return false;
+        }
+    }
+    return true;
+}
+
```
Comment (damla) [inline chapter01/1.2 - Check Perm/checkPermute.js:51]:
```suggestion

// Tests
```

File: chapter01/1.2 - Check Perm/checkPermute.js [hunk 4]
```diff
@@ -16,4 +16,39 @@ console.log(checkPermute('aba', 'aab'), true);
...
 console.log(checkPermute('aba', 'aaba'), false);
 
-console.log(checkPermute('aba', 'aa'), false);
\ No newline at end of file
+console.log(checkPermute('aba', 'aa'), false);
+
+// Same function as a one line solution
+
+checkPermuteOneLine = (a, b) => {
+    return (a.length === b.length && (a.split("").sort().join("") === b.split("").sort().join("")))
```
Comment (profnandaa) [inline chapter01/1.2 - Check Perm/checkPermute.js:24]: Unfortunately using inbuilt function. In most interviews, you'd be required to write that logic, unless it's just a utility, not part of the solution.

**Pull request closed without merging**